snapshot/linux/thread_snapshot_linux.cc

```diff
@@ -199,6 +205,36 @@ bool ThreadSnapshotLinux::Initialize(ProcessReaderLinux* process_reader,
 
   thread_id_ = thread.tid;
 
+#ifdef CLIENT_STACKTRACES_ENABLED
+  void *upt = _UPT_create(thread_id_);
+  if (upt) {
+    unw_addr_space_t as = unw_create_addr_space(&_UPT_accessors, __LITTLE_ENDIAN);
+    unw_cursor_t cursor;
+    if (unw_init_remote(&cursor, as, upt) == UNW_ESUCCESS) {
+      do {
+        unw_word_t addr;
+        if (unw_get_reg(&cursor, UNW_REG_IP, &addr) < 0) {
+          return false;
+        }
+
+        std::string sym("");
+        char buf[1024];
+        unw_word_t symbol_offset;
+        if (unw_get_proc_name(&cursor, buf, sizeof(buf), &symbol_offset) ==
+            UNW_ESUCCESS) {
```

Comment:
from the manpage of `unw_get_proc_name`:
```
UNW_ENOMEM
               The procedure name is too long to fit in the buffer provided. A truncated version
               of the name has been returned.
```
so i think this condition check is too simplistic, I think we should choose one of two choices:
* Accept truncated names
* Grow the buffer, up to some limit, and try again if the name fits in the new buffer.

Reply:
The problem is that the function does not give us any indication of how big the buffer is supposed to be. Which would mean this gets complicated, as you would have to try doubling the buffer size as long as you succeed.
IMO this is not worth the effort and we might as well ignore that case. Also, we have a 1K buffer, so that should cover all the realistic function names, except for extreme cases I don’t think we should cater for.